src/client/src/rt-theme/ThemeState.tsx

```diff
@@ -0,0 +1,96 @@
+import React from 'react'
+import { ThemeProvider } from './ThemeProvider'
+import { Theme, themes } from './themes'
+
+type ThemeName = 'light' | 'dark'
+
+interface ThemeSelector {
+  name?: ThemeName
```

Comment:
Are you sure this should be optional?

Reply:
It should not be optional in the provider. For the consumer it is optional.

Reply:
I agree with this

Reply:
`null` is a valid value, at which point you'd use the previous or root.

Reply:
Can you see a use case where you would need multiple providers for this code?